### PR TITLE
feat: add flush parameter to dbmethod decorator

### PR DESCRIPTION
**Pull request by NicolasOuihelp**
### **Contexte**
- Add flush param
- Add associated tests

### **Lien de Test Front**
- [ ] Pas de lien de test
- [ ] Back non merger, besoin de l'api local -> spécifier le chemin d'accès
- [ ] Directement utilisable -> spécifier le lien direct

<em>Lien/chemin: STP Lien direct de la page de test ou chemin d'accès</em>

### **Type de changement**
<em>STP coche les options pertinentes</em>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
### **Checklist avant de passer en `to review`**
- [ ] J'ai nettoyé les `debugger`, et autres `console.log`
- [ ] Le build est-il au vert (flow, tsc, eslint) ?
- [ ] J'ai écrit/modifié des tests explicites en particulier sur les edge cases
- [ ] J'ai commenté mon code
- [ ] J'ai respecté les [coding guidelines Ouihelp](https://www.notion.so/ouihelp/Ouihelp-2050-42577e63cb2d4d739cde9007b4ba94c5)
- [ ] J'ai correctement taggé ma branche (notamment cross dependency project)
- [ ] J'ai reviewé mon code
- [ ] J'ai testé les flows de ma feature / bug fix

### CODE REVIEW

File: tests/test_database.py [hunk 1]
```diff
@@ -36,3 +41,11 @@ def test_db_method_with_commit_true(self):
...
         self.worker.fake_method(commit=False)
         self.assertTrue(not self.dbsession.has_been_committed)
+
+    def test_db_method_with_flush_true(self):
+        self.worker.fake_method(flush=True)
+        self.assertTrue(self.dbsession.has_been_flushed)
+
+    def test_db_method_with_flush_false(self):
+        self.worker.fake_method(flush=False)
+        self.assertTrue(not self.dbsession.has_been_flushed)
```
Comment (nico-deforge) [inline tests/test_database.py]: assert False ?

**Review by nico-deforge — APPROVED** [review]
LGTM merci Nico !
Une petite remarque optionnelle sur un assert

**Review by jidey43 — APPROVED** [review]
Est-ce qu'on veut vérifier qu'on ne fait pas commit+flush ? c'est pas "grave" mais ça veut dire que celui qui appelle avec les deux flags ne sait pas ce qu'il fait

**Comment by nico-deforge** [comment]
> Est-ce qu'on veut vérifier qu'on ne fait pas commit+flush ? c'est pas "grave" mais ça veut dire que celui qui appelle avec les deux flags ne sait pas ce qu'il fait

Yes d'ailleurs si tu appelles flush=True et oublie commit=False, on a un comportement bizarre non ? ça va dans le sens de la remarque de Roman.
